src/parser/jobs/dnc/modules/Technicalities.tsx

```diff
@@ -0,0 +1,262 @@
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+import Timeline from 'parser/core/modules/Timeline'
+
+import DISPLAY_ORDER from './DISPLAY_ORDER'
+import FeatherGauge from './FeatherGauge'
+
+// Harsher than the default since you'll only have 4-5 total windows anyways
+const TECHNICAL_SEVERITY_TIERS = {
+	1: SEVERITY.MINOR,
+	2: SEVERITY.MEDIUM,
+	3: SEVERITY.MAJOR,
+}
+
+const WINDOW_STATUSES = [
+	STATUSES.DEVILMENT.id,
+	STATUSES.TECHNICAL_FINISH.id,
+]
+
+const FEATHER_THRESHHOLD = 3
+const POST_WINDOW_GRACE_PERIOD_MILLIS = 500
+
+class TechnicalWindow {
+	start: number
+	end?: number
+
+	rotation: Array<AoeEvent | CastEvent> = []
+	gcdCount: number = 0
+	trailingGcdEvent?: CastEvent
+
+	hasDevilment: boolean = false
+	timelyDevilment: boolean = true
+	poolingProblem: boolean = false
+
+	constructor(start: number) {
+		this.start = start
+	}
+}
+
+export default class Technicalities extends Module {
+	static handle = 'technicalities'
+	static title = t('dnc.technicalities.title')`Technical Windows`
+	static displayOrder = DISPLAY_ORDER.TECHNICALITIES
+
+	@dependency private combatants!: Combatants
+	@dependency private suggestions!: Suggestions
+	@dependency private timeline!: Timeline
+	@dependency private feathers!: FeatherGauge
+
+	private history: TechnicalWindow[] = []
+	private firstDevilment: boolean = false
+	private badDevilments: number = 0
+
+	protected init() {
+		this.addHook('applybuff', {to: 'player', abilityId: STATUSES.TECHNICAL_FINISH.id}, this.tryOpenWindow)
+		this.addHook('removebuff', {to: 'player', abilityId: WINDOW_STATUSES}, this.tryCloseWindow)
+		this.addHook('cast', {by: 'player'}, this.onCast)
+		this.addHook('complete', this.onComplete)
+	}
+
+	private tryOpenWindow(event: BuffEvent) {
+		const lastWindow: TechnicalWindow | undefined = _.last(this.history)
+
+		// Handle multiple dancer's buffs overwriting each other, we'll have a remove then an apply with the same timestamp
+		// If that happens, re-open the last window and keep tracking
+		if (lastWindow) {
+			if (!lastWindow.end) {
+				return
+			}
+			if (lastWindow.end === event.timestamp) {
+				lastWindow.end = undefined
+				return
+			}
+		}
+
+		const newWindow = new TechnicalWindow(event.timestamp)
+		this.history.push(newWindow)
+	}
+
+	private tryCloseWindow(event: BuffEvent) {
+		const lastWindow: TechnicalWindow | undefined = _.last(this.history)
+
+		if (!lastWindow) {
+			return
+		}
+
+		if (this.isWindowOkToClose(lastWindow, event)) {
+			lastWindow.end = event.timestamp
+
+			// Check to see if this window could've had more feathers due to possible pooling problems
+			if (this.feathers.feathersSpentInRange(lastWindow.start, lastWindow.end) < FEATHER_THRESHHOLD) {
+				const previousWindow = this.history[this.history.length-2]
+				const feathersBeforeWindow = this.feathers.feathersSpentInRange((previousWindow ? previousWindow.start : this.parser.fight.start_time)
```

Comment:
Should this be `previousWindow.end`? potentially going to pick up spent feathers in the previous window's duration as-is, i _think_.

Reply:
Hurrr, yes, you're right. Don't code while MSQlette-ing, kids. I'll fix that tonight.